addons/website_slides_survey/models/survey_user_input.py

```diff
@@ -12,16 +11,35 @@ class SurveyUserInput(models.Model):
         help="The related course slide when there is no membership information")
     slide_partner_id = fields.Many2one('slide.slide.partner', 'Subscriber information',
         help="Slide membership information for the logged in user",
-        index='btree_not_null') # index useful for deletions in comodel
+        index='btree_not_null')  # index useful for deletions in comodel
+
+    @api.depends('survey_id.certification')
+    def _compute_can_access_survey_user_input(self):
+        return super()._compute_can_access_survey_user_input()
+
+    def _get_access_domain(self):
+        domain = super()._get_access_domain()
+        if not self.env.user.has_group("website_slides.group_website_slides_officer"):
+            return domain
+        domain_slides = self._get_access_domain_website_slides_survey()
+        return expression.OR([domain, domain_slides])
+
+    @api.model
+    def _get_access_domain_website_slides_survey(self):
```

Comment:
same for this one (it will make the code less complex, and we can still split it if needed later)